src/mlpack/methods/linear_regression/linear_regression_predict_main.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file methods/linear_regression/linear_regression_predict_main.cpp
+ * @author James Cline
+ *
+ * Main function for least-squares linear regression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/io.hpp>
+
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME linear_regression_predict
+
+#include <mlpack/core/util/mlpack_main.hpp>
+
+#include "linear_regression.hpp"
+
+using namespace mlpack;
+using namespace mlpack::regression;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+// Program Name.
+BINDING_USER_NAME("Linear Regression Prediction");
+
+// Short description.
+BINDING_SHORT_DESC("Predictions from model.");
+
+// Long description.
+BINDING_LONG_DESC("");
+
+// Example.
+BINDING_EXAMPLE(
+  CALL_METHOD("model", "predict", "test", "X_test"));
+
+PARAM_MODEL_IN(LinearRegression, "input_model", "Existing LinearRegression "
+    "model to use.", "m");
+
+PARAM_MATRIX_IN("test", "Matrix containing X' (test regressors).", "T");
+
+// This is the future name of the parameter.
+PARAM_ROW_OUT("output_predictions", "If --test_file is specified, this "
+    "matrix is where the predicted responses will be saved.", "o");
+
+void BINDING_FUNCTION(util::Params& params, util::Timers& timer)
+{
+  LinearRegression* lr;
+	RequireOnlyOnePassed(params, {"input_model"}, true); // input_model must be passed.
+	RequireOnlyOnePassed(params, {"test"}, true); // test matrix must be passed.
+
+  // A model file was passed in, so load it.
+  timer.Start("load_model");
+  lr = params.Get<LinearRegression*>("input_model");
```

Comment:
```suggestion
  LinearRegression* lr = params.Get<LinearRegression*>("input_model");
```
This happens a couple times in the adapted bindings, but you could often declare and set the value of a model in the same line for simplicity. :+1: